spec/RedisCacheAdapter.spec.js

```diff
@@ -0,0 +1,41 @@
+var RedisCacheAdapter = require('../src/Adapters/Cache/RedisCacheAdapter').default;
+
+describe('RedisCacheAdapter', function() {
+  var KEY = 'hello';
+  var VALUE = 'world';
+
+  function wait(sleep) {
+    return new Promise(function(resolve) {
+      setTimeout(resolve, sleep);
+    })
+  }
+
+  it('should get/set/clear', (done) => {
+    var cache = new RedisCacheAdapter({
+      ttl: NaN
+    });
+
+    cache.put(KEY, VALUE)
+      .then(() => cache.get(KEY))
+      .then((value) => expect(value).toEqual(VALUE))
```

Comment:
what!@?! mind blown.  I don't even understand how returning expect.... is "thenable", but super cool idiom.  going to start using....